lms/models/application_settings.py

```diff
@@ -25,7 +22,10 @@ def set(self, group, key, value):
         :param key: The key in that group
         :param value: The value to set
         """
-        self.data.setdefault(group, {})[key] = value
+        super().setdefault(group, {})[key] = value
+
+    def __repr__(self):
+        return f"ApplicationSettings({super().__repr__()})"
 
-    def clone(self):
-        return ApplicationSettings(deepcopy(self.data))
```

Comment:
The clone method is no longer needed, you can just use `copy.deepcopy` directly:

```python
settings_copy = copy.deepcopy(ai.settings)
```

`settings` will be a new `ApplicationSettings` object